public_html/wp-content/plugins/camptix/addons/require-login.php

```diff
@@ -71,16 +71,22 @@ public function block_unauthenticated_actions() {
 			return;
 		}
 
-		// Temporary: We don't want to block users from editing tickets.
-		// See: https://github.com/WordPress/wordcamp.org/issues/1393.
-		if ( ! is_user_logged_in() && ! $this->user_is_editing_ticket() ) {
+		if ( ! is_user_logged_in() ) {
+
+			// Temporary: We don't want to block users from editing tickets unless they are unconfirmed.
+			// See: https://github.com/WordPress/wordcamp.org/issues/1393.
+			// See: https://github.com/WordPress/wordcamp.org/issues/1420.
+			if ( $this->user_is_editing_ticket() && ! $this->user_must_confirm_ticket( $_REQUEST['tix_attendee_id'] ) ) {
+				return;
+			}
+
 			$args = array();
-			// If this was a registration, pass through the selected tickets and coupon.
-			if ( 'attendee_info' === $_REQUEST['tix_action'] && isset( $_REQUEST['tix_tickets_selected'] ) ) {
-				$args['tix_action'] = $_REQUEST['tix_action'];
-				$args['tix_tickets_selected'] = $_REQUEST['tix_tickets_selected'];
-				if ( isset( $_REQUEST['tix_coupon'] ) ) {
-					$args['tix_coupon'] = $_REQUEST['tix_coupon'];
+			if ( in_array( $_REQUEST['tix_action'], array( 'attendee_info', 'edit_attendee' ) ) ) {
+				// Pass along all `tix_` information.
+				foreach ( $_REQUEST as $key => $value ) {
```

Comment:
we'll want to be explicit in passing the values from $_REQUEST (i.e. checking they are what we expect) and checking their value before adding to `add_query_arg`

Reply:
Sure. I can do that. How about the changes in [5346a21](https://github.com/WordPress/wordcamp.org/pull/1421/commits/5346a2122cc0d528f8f3717b784e93bd412ece66)?

Reply:
Nice!